addons/account/models/account_move_line.py

```diff
@@ -2631,6 +2640,7 @@ def _prepare_exchange_difference_move_vals(self, amounts_list, company=None, exc
                     'debit': -amount_residual if amount_residual < 0.0 else 0.0,
                     'credit': amount_residual if amount_residual > 0.0 else 0.0,
                     'amount_currency': -amount_residual_currency,
+                    'full_reconcile_id': line.full_reconcile_id.id,
```

Comment:
This doesn't seem to be related to the import of matching numbers....
shouldn't there be a test to check the matching numbers of all reconciled lines including the exchange diff

Reply:
I added this line because of a failing test (see the first batch)
Basically, it is adding a exchange difference on the full reconcile after it was created; but this is only rarely done.
Basically that failing test is now checking what you asked too, thanks to the constraint :laughing: